### ISSUE TITLE
Add end-to-end tests for localization

### ISSUE THREAD

**Issue by BenHenning** [oppia/oppia-android#3839]
Localization support (i.e. post #3797) introduces a complex, multi-activity mechanism for making sure that language selection works in a multitude of scenarios. Since it's multi-activity, both Robolectric & Espresso aren't ideal for testing the mechanism in its entirety. Instead, UiAutomator end-to-end tests should be added to verify that changing the system language correctly recreates & changes the language in-app.

**Comment by justdvnsh** [comment]
@BenHenning @rt4914 I'd like to work on this issue

**Comment by BenHenning** [comment]
Hi @justdvnsh. I'm glad to see you're interested in this issue.

Note that this issue is effectively blocked on #3866 (& all of its previous PRs). After those are merged, I think this PR can be worked on. However, some things to keep in mind:
- Our end-to-end tests don't have fully-established documentation yet, so you'll need to write them based primarily on existing code references, plus https://github.com/oppia/oppia-android/wiki/End-to-End-Testing-Guide exists which will be a big help when running/building tests
- You'll need to spec out the end-to-end tests based on the scenarios that you think are worth testing (I'm happy to confirm these before you get too deeply into implementation work)
- The end-to-end tests do not run in Gradle (so you'll need a working Bazel environment locally--see the wiki for setting this up if it isn't already)
- The end-to-end tests don't run correctly in Bazel today, but they can be built & manually deployed/started

Given the above disclaimers, are you still interested in working on this? :) Just want to make sure since this might be a harder issue than ones you've worked on in the past.

**Comment by justdvnsh** [comment]
@BenHenning Yup. I'd still want to do it. The contributions I have made were to get involved me in this organization. I now want to make more important contributions and become a core member of the foundation. So, yeah I'd like to work upon this. Moreover, I am anyhow working on my testing area for improvement, so I think this would be a great start. By the time this issue get unblocked , I'd like to take up #3846 .
